### PR TITLE
fixes #174

### LINKED ISSUE #174: lint: add checker for unnessesary blocks
Warn on blocks without local variables.
**before:**
```go
a := 5
{
     fmt.Print(a)
}
```
**after:**
```go
a := 5
fmt.Print(a)
```

### PR DESCRIPTION
**Pull request by dmirogin**

### CODE REVIEW

**Comment by dmirogin** [comment]
Could somebody help me ?
Why test fail here https://travis-ci.org/go-critic/go-critic/jobs/406399595 ? It'is positive, it must be warned, right? What am i doing wrong ?

File: lint/testdata/unnecessaryBlock/positive_tests.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package checker_test
+
+func positive () {
+	a := 1
+	{
```
Comment (cristaloleg) [inline lint/testdata/unnecessaryBlock/positive_tests.go]: Here you should have a comment like:
```block doesn't have assignment statements, can be simply deleted```

**Comment by cristaloleg** [comment]
@dmirogin nicely done. Just add a comment and the build will be green 😉

**Comment by dmirogin** [comment]
@cristaloleg Cool. Thanks 👍

**Comment by Quasilyte** [comment]
Note that we're using https://golang.org/doc/contribute.html#commit_messages as a guidance for commit message structure. Squashed commit has updated message to comply to that.

**Comment by cristaloleg** [comment]
@dmirogin thanks for the contribution 😉

**Pull request merged**